### ISSUE TITLE
Feature request: equivalent of SC2206 and SC2207 for `for`-loops

### ISSUE THREAD

**Issue by LukeShu** [koalaman/shellcheck#894]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Shellcheck intentionally doesn't warn about `$list` being unquoted in `for x in $list;`, as it is often used to intentionally split a string.  However, it is my experience that most developers using it this way don't realize that it *also* performs glob expansion; it isn't a safe way to perform the splitting that developers expect it to perform.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
list='foo bar * baz'
for x in $list; do
	echo "$x"
done
```
I believe that most developers would expect the output of the above program to be

```
foo
bar
*
baz
```

This is not the case; as it expands the `*` as a glob.

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

```
Line 3:
for x in $list; do
         ^-- SC????: Prefer mapfile or read -a to split robustly
```

I would like shellcheck to force me to write something like

```
read -a _list <<<"$list"
for x in "${_list[@]}"; do
```
